server/api/index.js

```diff
@@ -264,4 +265,48 @@ router.get("/users", (req, res) => {
   });
 });
 
+router.get("/user-roles", async (req, res) => {
+  const data = await db.getUsersByRole();
+  const newData = data.reduce((result, current) => {
+    const userId = current.user_id;
+    if (result[userId]) {
+      result[userId]["roles"].push(current.role);
+    } else {
+      result[userId] = {
+        name: current.name,
+        user_id: current.user_id,
+        email: current.email
+      };
+      result[userId]["roles"] = [current.role];
+    }
+
+    return result;
+  }, {});
+
+  res.send(Object.values(newData));
+});
+
+router.get("/user-roles/:id", async (req, res) => {
+  const userId = req.params.id;
+  const data = await db.getUserRoles(userId);
+  res.send(data);
+});
+
+// Add user roles to the user_roles table
+router.post("/user-roles", async (req, res) => {
+  const body = req.body;
+  try {
+    await db.clearRolesByUser(body.user_id)
```

Comment:
this can be abstracted in one method in dbHelper .. just call it assignRolesToUser .. and let it do the clearing, then assigning ... no need to expose `clearRolesByUser` through the module - nothing else will use it, right? and it's pretty dangerous one

Reply:
Also this avoid an issue with this code .. doing forEach (which synchronous) then calling an async operation ... right now, this code always sends back `successfully assigned roles` even if one the `db.addRoleToUser` fails

Reply:
then knex.insert can take an array .. no need for looping and inserting